common/Behaviors/RenderWebHyperlinksBehavior.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Documents;
+using Microsoft.Xaml.Interactivity;
+
+namespace DevHome.Common.Behaviors;
+
+/// <summary>
+/// A behavior that renders http and https hyperlinks in a <see cref="TextBlock"/>.
+/// </summary>
+public partial class RenderWebHyperlinksBehavior : Behavior<TextBlock>
+{
+    protected override void OnAttached()
+    {
+        base.OnAttached();
+        AssociatedObject.Loaded += OnLoaded;
+    }
+
+    protected override void OnDetaching()
+    {
+        base.OnDetaching();
+        AssociatedObject.Loaded -= OnLoaded;
+    }
+
+    private void OnLoaded(object sender, RoutedEventArgs e)
+    {
+        SetInlineCollection(ToInline(AssociatedObject.Text));
+    }
+
+    private List<Inline> ToInline(string text)
+    {
+        List<Inline> elements = [];
+        if (string.IsNullOrEmpty(text))
+        {
+            return elements;
+        }
+
+        foreach (var part in WebUrlRegex().Split(text))
+        {
+            if (Uri.IsWellFormedUriString(part, UriKind.Absolute))

```

Comment:
https://github.com/dotnet/runtime/issues/72632 IsWellFormedString can return false positives.  The original message contains a work around.

Reply:
I replaced `IsWellFormedUriString` with `TryCreate`